front_end/content.py

```diff
@@ -586,9 +587,12 @@ def get_registered_courses(self, user_id):
 
         return registered_courses
 
-    def get_partner_info(self, course, user_id):
+    def get_partner_info(self, course, user_id, include_self=False):
         # Gets list of users.
-        users = [x[1] for x in self.get_registered_students(course) if not x[0] == user_id]
+        if include_self:
+            users = [x[1] for x in self.get_registered_students(course)]
+        else:
+            users = [x[1] for x in self.get_registered_students(course) if not x[0] == user_id]
 
         # Adds users to dict to find duplicate names.
         user_duplicates_dict = {}
```

Comment:
One of my recent commits adds an email_address field to the users database table. So we are no longer using email addresses as the user id (well, in some cases we are not). So please modify this to pull the email address from the email_address field rather than the id field.

Please do the same anywhere else in the code where this applies. I'm thinking of the part of the code where students indicate who their pair programming partner is.